docs/guide/in-depth/advanced-chunks.md

```diff
@@ -1,6 +1,238 @@
+`AdvancedChunks` is a powerful feature that allows you to do `manual chunking` against the `automatic chunking` done by [code splitting](./code-splitting.md). This is useful when you want to optimize the loading performance of your application by splitting it into smaller, more manageable pieces.
+
+Before reading this guide, you should first understand the [code splitting](./code-splitting.md) feature of Rolldown. This guide will explain how `advancedChunks` works and how to use it effectively.
+
+Before we dive into the details, let's clarify some things first.
+
+- `automatic chunking` and `manual chunking` are not contradictory. Using `manual chunking` does not mean disabling `automatic chunking`.
+- A module will be either captured by `automatic chunking` or `manual chunking` depending on your configuration, but not both. If a module is not captured by `manual chunking`, it will be still put into a chunk which is created by `automatic chunking` with respecting rules we explained in the [code splitting](./code-splitting.md) guide.
+
+## Why use `advancedChunks`?
+
+The `automatic chunking` doesn't take loading performance or cache invalidation into account. It simply groups modules based on their static imports. This can lead to suboptimal chunking, where large chunks are created that may not be performant for loading or cause cache invalidation for every deployment.
+
+## How to use `advancedChunks`?
+
+Let's take a look at the following example:
+
+```jsx
+// index.jsx
+import * as ReactDom from 'react-dom';
+import App from './App.jsx';
+
+ReactDom.createRoot(document.getElementById('root')).render(
+  <App />,
+);
+
+// App.jsx
+import * as React from 'react';
+import { Button } from 'ui-lib';
+
+export default function App() {
+  return <Button onClick={() => alert('Button clicked!')} />;
+}
+```
+
+and you get the following output:
+
+```js [output-hash0.js]
+// node_modules/react/index.js
+'React library code';
+
+// node_modules/ui-lib/index.js
+'UI library code';
+
+// node_modules/react-dom/index.js
+'ReactDOM library code';
+
+// App.js
+function App() {
+  return <Button onClick={() => alert('Button clicked!')} />;
+}
+
+// index.js
+
+ReactDom.createRoot(document.getElementById('root')).render(<App />);
+```
+
+In this example,
+
+- We used 3 libraries: `react`, `react-dom`, and `ui-lib`.
+- `output-hash0.js` is the output file generated by Rolldown.
+- `hash0` is the hash of the output file, it changes if the content of the file changes.
+
+### Reduce cache invalidation
+
+Let's talk about cache invalidation first. Cache invalidation here means that when you deploy a new version of your application, the browser will need to download the new version of the file. If the file is large, this can lead to a poor user experience.
```

Comment:
I think we need to mention the `Cache-Control` header somewhere as we assume that is set.

Reply:
Maybe. That will make this guide more academic. I assume people whoe read this have this kind of knowledge. But definitely it's possitive to add such information.